src/jwt/jwtparser.c

```diff
@@ -59,13 +59,19 @@ static cjose_jwk_t *get_jwk(JWS *jws)
     if (!doc)
         goto errorExit;
 
-    keyid = DIDURL_FromString(JWS_GetKeyId(jws), NULL);
-    if (!keyid)
-        goto errorExit;
-
-    key = DIDDocument_GetPublicKey(doc, keyid);
-    if (!key)
-        goto errorExit;
+    if (!JWS_GetKeyId(jws)) {
+        keyid = DIDDocument_GetDefaultPublicKey(doc);
+        key = DIDDocument_GetPublicKey(doc, keyid);
+        if (!key)
+            goto errorExit;
+    }
+    else {
+        keyid = DIDURL_FromString(JWS_GetKeyId(jws), issuer);
+        key = DIDDocument_GetPublicKey(doc, keyid);
+        DIDURL_Destroy(keyid);
+        if (!key)
+            goto errorExit;
+    }
```

Comment:
Or we write these ones:
  if (!JWS_GetKeyId(jws)) {
      ...
  } else {
      ...
  }
  DIDURL_Destroy(keyid);
  if (!key) 
         goto errorExit;

Reply:
> Or we write these ones:
> if (!JWS_GetKeyId(jws)) {
> ...
> } else {
> ...
> }
> DIDURL_Destroy(keyid);
> if (!key)
> goto errorExit;

no,  one branch is not destroy 'keyed'.